src/token/token.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* eslint-env browser */
+
+'use strict';
+
+import mix from '@ckeditor/ckeditor5-utils/src/mix';
+import ObservableMixin from '@ckeditor/ckeditor5-utils/src/observablemixin';
+
+const DEFAULT_OPTIONS = { refreshIntervalTime: 3600000, startAutoRefresh: true };
+
+/**
+ * Class representing the token used for communication with CKEditor Cloud Services.
+ * Value of the token is retrieving from the specified URL and is refreshed every 1 hour by default.
+ *
+ * @mixes ObservableMixin
+ */
+class Token {
+	/**
+	 * Creates `Token` instance.
+	 *
+	 * @param {String} tokenUrl Endpoint address to download the token.
+	 * @param {String} [initTokenValue] Initial value of the token.
+	 * @param {Object} options
+	 * @param {Number} [options.refreshIntervalTime=3600000] Delay between refreshes. Default 1 hour.
+	 * @param {Boolean} [options.startAutoRefresh=true] Specifies whether to start the refresh automatically.
+	 */
+	constructor( tokenUrl, initTokenValue, options = DEFAULT_OPTIONS ) {
```

Comment:
Since `initTokenValue` is optional, could we move it to the options object? It makes constructor more complicated and you need to put `null` if you want to pass `options` without init value. Also, I would rename it to `initValue`, since it's obvious that it's a token value.